Assets/MixedRealityToolkit.SDK/Experimental/Features/Dwell/DwellHandler.cs

```diff
@@ -0,0 +1,243 @@
+                {
+                    DwellIntended.Invoke(pointer);
+                    CurrentDwellState = DwellStateType.DwellIntended;
+                }
+                else if (CurrentDwellState == DwellStateType.DwellIntended && (focusDuration - dwellProfile.DwellIntentDelay) >= dwellProfile.DwellStartDelay)
+                {
+                    DwellStarted.Invoke(pointer);
+                    CurrentDwellState = DwellStateType.DwellStarted;
+                    FillTimer = dwellProfile.TimeToCompleteDwell;
+                }
+                else if (CurrentDwellState == DwellStateType.DwellStarted && (focusDuration - dwellProfile.DwellIntentDelay - dwellProfile.DwellStartDelay) >= dwellProfile.TimeToCompleteDwell)
+                {
+                    DwellCompleted.Invoke(pointer);
+                    CurrentDwellState = DwellStateType.DwellCompleted;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Default FillTimer computation based on profile settings
+        /// </summary>
+        protected virtual void UpdateFillTimer()
+        {
+            switch (CurrentDwellState)
+            {
+                case DwellStateType.None:
+                case DwellStateType.FocusGained:
+                case DwellStateType.DwellIntended:
+                    FillTimer = 0;
+                    break;
+                case DwellStateType.DwellStarted:
+                    FillTimer -= Time.deltaTime;
+                    FillTimer = FillTimer < 0 ? 0 : FillTimer;
+                    break;
+                case DwellStateType.DwellCompleted:
+                    Debug.Assert(FillTimer < float.Epsilon, $"Fill timer was not 0. It was {FillTimer}.");
+                    break;
+                case DwellStateType.DwellCanceled:
+                    if ((DateTime.UtcNow - focusExitTime).TotalSeconds > dwellProfile.TimeToAllowDwellResume)
+                    {
+                        FillTimer = 0;
+                        CurrentDwellState = DwellStateType.None;
+                    }
+                    break;
+                default:
+                    FillTimer = 0;
+                    break;
+            }
+        }
+
+        public void OnFocusEnter(FocusEventData eventData)
+        {
+            if (eventData.NewFocusedObject == gameObject
+                && eventData.Pointer.InputSourceParent.SourceType == dwellProfile.DwellPointerType)
+            {
+                HasFocus = true;
+
+                // check intent to resume
+                if (CurrentDwellState == DwellStateType.DwellCanceled
+                    && pointer.InputSourceParent.SourceId == eventData.Pointer.InputSourceParent.SourceId //make sure the returning pointer id is the same
+                    && (DateTime.UtcNow - focusExitTime).TotalSeconds <= dwellProfile.TimeToAllowDwellResume)
+                {
+                    focusEnterTime = focusEnterTime.AddSeconds(dwellProfile.TimeToAllowDwellResume);
```

Comment:
>focusEnterTime = focusEnterTime.AddSeconds(dwellProfile.TimeToAllowDwellResume); [](start = 20, length = 80)

Can you add a comment about why focusEnterTime isn't just when OnFocusEnter called?